Modify rule S106: Migrate to LayC - Standard outputs should not be used directly to log anything (#3280)
SonarSource/rspec: rules/S106/description.adoc

In software development, logs serve as a record of events within an application, providing crucial insights for debugging.

> When logging, it's essential to ensure that the logs are:

Reviewer: We should avoid contractions
```suggestion
When logging, it is essential to ensure that the logs are:
```

* easily accessible
* uniformly formatted for readability
* properly recorded
* securely logged when dealing with sensitive data

> If a program directly writes to the standard outputs (i.e. {language_std_outputs}) those requirements are not met.

Reviewer: Grammarly suggests swapping the two components of the sentence. I think it helps.

Another small nit: the list of std outputs for C++ is not exaustive. It should be "e.g.," (with a comma) instead of "i.e.". I suppose it's also not exaustive for other languages.

```suggestion
Those requirements are not met if a program directly writes to the standard outputs (e.g., {language_std_outputs}).
```

> That's why defining and using a dedicated logger is highly recommended.

Reviewer: ```suggestion
That is why defining and using a dedicated logger is highly recommended.
```
